findSurfaceCharge.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import print_function
+from pymol import cmd
+
+
+def findSurfaceAtoms(selection="all",cutoff=2.5):
+    """
+    Adapted from Jason Vertrees https://pymolwiki.org/index.php/FindSurfaceResidues
+DESCRIPTION
+
+    Finds those atoms on the surface of a protein
+    that have at least 'cutoff' exposed A**2 surface area.
+
+USAGE
+
+    findSurfaceAtoms [ selection, [ cutoff ]]
+
+SEE ALSO
+
+    findSurfaceResidues
+    """
+    cutoff = float(cutoff)
+
+    tmpObj = cmd.get_unused_name("_tmp")
+    cmd.create(tmpObj, "(" + selection + ") and polymer", zoom=0)
+
+    cmd.set("dot_solvent", 1, tmpObj)
+    cmd.get_area(selection=tmpObj, load_b=1)
+
+    # threshold on what one considers an "exposed" atom (in A**2):
+    cmd.remove(tmpObj + " and b < " + str(cutoff))
+
+    selName = cmd.get_unused_name("exposed_atm_")
+
+    cmd.select(selName, "(" + selection + ") in " + tmpObj)
+
+    cmd.delete(tmpObj)
+
+    return selName
+
+
+def _findSurfaceChargeImpl(selection, pH, folded, cutoff):
+
+    def get_exposed_residues(selection,cutoff):
+        cutoff = float(cutoff)
+
+
+        selName = findSurfaceAtoms(selection, cutoff)
+
+        tempExposed = set()
+        cmd.iterate(selName, "tempExposed.add((resv,oneletter))", space=locals())
```

Comment:
This might not work as expected with multi-chain proteins. To identify a residue, the key should be `(model, segi, chain, resv, oneletter)`.

Reply:
Or better: `resi` instead of `resv` to also cover insertion codes.

Reply:
I pushed a commit with this change